RichTextAdapter types do not strictly match on blank template

### Link to reproduction

https://github.com/joas8211/payload-editor-type-bug

### To Reproduce

#### From scratch
1. Create project from blank template.
    `npx create-payload-app payload-editor-type-bug -t blank`
2. Modify tsconfig.json to set `strict` to `true`.
3. Install and build.
    `yarn install && yarn build`

#### Using reproduction repository
1. Clone the repository.
    `git clone https://github.com/joas8211/payload-editor-type-bug`
2. Install and build.
    `yarn install && yarn build`

### Describe the Bug

When build is ran the following error is printed:
```
src/payload.config.ts:16:3 - error TS2322: Type 'RichTextAdapter<AdapterArguments>' is not assignable to type 'RichTextAdapter<unknown>'.
```

You can also see the error highlighted in VSCode. There also some other errors which are not strictly related to this issue.

### Payload Version

2.0.0

--- a/packages/payload/src/admin/components/forms/field-types/RichText/types.ts
+++ b/packages/payload/src/admin/components/forms/field-types/RichText/types.ts
@@ -9,7 +9,7 @@ export type RichTextFieldProps<AdapterProps = unknown> = Omit<
   path?: string
 }
 
-export type RichTextAdapter<AdapterProps = unknown> = {
+export type RichTextAdapter<AdapterProps = object> = {
   CellComponent: React.FC<CellComponentProps<RichTextField<AdapterProps>>>
   FieldComponent: React.FC<RichTextFieldProps<AdapterProps>>
   afterReadPromise?: (data: {

--- a/packages/payload/src/admin/components/forms/field-types/RichText/types.ts
+++ b/packages/payload/src/admin/components/forms/field-types/RichText/types.ts
@@ -2,7 +2,7 @@ import type { PayloadRequest } from '../../../../../express/types'
 import type { RichTextField, Validate } from '../../../../../fields/config/types'
 import type { CellComponentProps } from '../../../views/collections/List/Cell/types'
 
-export type RichTextFieldProps<AdapterProps = unknown> = Omit<
+export type RichTextFieldProps<AdapterProps = object> = Omit<
   RichTextField<AdapterProps>,
   'type'
 > & {

--- a/packages/payload/src/fields/config/types.ts
+++ b/packages/payload/src/fields/config/types.ts
@@ -398,7 +398,7 @@ export type RelationshipValue =
   | ValueWithRelation[]
   | (number | string)
 
-export type RichTextField<AdapterProps = unknown> = FieldBase & {
+export type RichTextField<AdapterProps = object> = FieldBase & {
   admin?: Admin
   editor?: RichTextAdapter<AdapterProps>
   type: 'richText'

